### ISSUE TITLE
Include text related to TLS library issues

### ISSUE THREAD

**Issue by toerless** [anima-wg/anima-brski-prm#82]
From todays BRSKI meeting discussion, i learned:

registrar-agents should be supported on mobile devices such as (i guess) iOS/Android/Windows portable devices - smartphones/tablets/notebooks

The TLS libraries often available on these OS's do only allow to verify certificates if these are web-certificates, e.g.: if the certificate includes some IP-address and/or DNS-name identity that can be matched, and (i am also guessing) are being signed by a WebPKI CA (for which the OS has the mechanisms in place to retrieve/update the CA).

If instead the server has only a private PKI, it may be potentially necessary to include into an app a non-OS based TLS library, which is highly undesirable.

PRM relies on TLS for the registrar-agent (as initiator) connect to a registrar (as TLS/HTTPs server). In BRSKI, there is no expectation for the registar to have a WebPKI certificate. Instead, the assumption was that it could/would only need to have a domain certificate, and the domain certificate in many domains, such as ACP or likely also many brski-prm use cases can only be a private certificate chain (see e.g.: explanation of requirements why this is so in RFC8994).

In conclusion i suggest:

I think the PRM draft should include text like this:

If the domain certificate chain is not a WebPKI chain, then the registrar SHOULD have an additional WebPKI cretificate (some RFC reference here would help) easily supported by the OS supported TLS libraries in the OSs intended to be used for registrar-agents. Registrar-agents MUST support configuration or discovery of registrar(s) by DNS name and DNS to resolve to connect to the registrar.

**Comment by stfries** [comment]
The verification of the certificate chain is nothing specific for PRM and applies to different protocols. 

I propose to take over the part as this is PRM-specific:
Registrar-agents MUST support configuration or discovery of registrar(s) by DNS name and DNS to resolve to connect to the registrar.

**Comment by stfries** [comment]
Given the current text we have in section 5.4.1: 
_The discovery of the domain registrar may be done as specified in {{RFC8995}} with the deviation that it is done between the registrar-agent and the domain registrar.
Alternatively, the registrar-agent may be configured with the address of the domain registrar and the certificate of the domain registrar._

I propose the following updated text considering Toerless' proposal: 
_The discovery of the domain registrar may be done as specified in {{RFC8995}} with the deviation that it is done between the registrar-agent and the domain registrar.
Alternatively, the registrar-agent may be configured with the address of the domain registrar and the certificate of the domain registrar. To support this, registrar-agents MUST support configuration of registrar(s) by a DNS name and DNS to resolve to connect to the registrar._

**Comment by mkovatsc** [comment]
This came up in the context why the registrar-agent does not utilize TLS when connecting to the pledge to trigger it.

The issue is that the pledge at this point only has an IDevID certificate, which does not contain any identifier to tie it to the connection being established such as a FQDN or IP address. Hence, the TLS handshake with all libraries available will fail (i.e., not providing any trust), and hence not providing any benefit other than hiding the HTTP content -- content that is pretty obvious to an eavesdropper anyway, not really giving much in addition.

One could write a special TLS library that works with IDevIDs, but the effort is very high, and hence rather hindering the adoption of BRSKI-PRM. Also, it would just replicate the trust establishment that is already done via the signed objects used by PRM.

**Comment by stfries** [comment]
The limitations for the TLS handshake are already stated in section 1 (and section 4): 

TLS with server-side, certificate-based authentication is not directly applicable, as the pledge only possesses an IDevID certificate. First, the IDevID does not contain any anchor for which any kind of [RFC6125] validation can be done. Second, the registrar-agent may not be aware of manufacturer trust anchors to validate the IDevIDs. Finally, IDevIDs do not typically set Extended Key Usage (EKU) for TLS WWW Server authentication.

Do we need to include further information to state that this functionality may not be available in standard TLS libraries?

**Comment by stfries** [comment]
Based on the discussion on the[ mailing list](https://mailarchive.ietf.org/arch/msg/anima/gE1XZWgXS9L84JhyWQ6ZoILbln0/)  the following is proposed: 

- Include optional TLS support for the communication between the registrar-agent and the pledge optionally to ensure privacy protection of this communication link. This allows the protection if necessary and keeps the object based security as used so far. 
 
This would require the following changes:
- Introduction: Change text to underline that the approach in BRSKI-PRM was defined independent of the underlying transport and to be used also via BTLE and NFC. Remove argumentation for TLS being not usable.
- Section 5.3: besides the mandatory requirement for the registrar-agent to possess the list with product-serial-numbers of pledges to be bootstrapped it SHOULD also possess the trustanchor for the IDevID verification
- Section 6.1 Clarify TLS option (trustanchor, certificate verification rules, ...) and update Figure 2 accordingly

**Comment by stfries** [comment]
Clarify proof of possession of the private key of the IDevID at the registrar-agent. Section 5.3 in the comment above.
Can be incorporated

**Comment by mcr** [comment]
> The TLS libraries often available on these OS's do only allow to verify certificates if these are web-certificates, e.g.: if the certificate includes some IP-address and/or DNS-name identity that can be matched, and (i am also guessing) are being signed by a WebPKI CA (for which the OS has the mechanisms in place to retrieve/update the CA).

It's not that they *can't* verify these certificates.
I'm saying that the OS makes it rather hard to do other than DNS-ID verification, the result being that many people writing code wind up *not* verifying the certificate at all.
I don't object to including https as an option, but I think that PRM needs to be analysed with the assumption that it's not there.

**Comment by stfries** [comment]
can be closed as optional TLS has been included. left open for Toerless to review resolution

**Comment by stfries** [comment]
closed as of design team meeting (August 1, 2023)
